pkg/apis/pipeline/v1alpha1/taskrun_validation.go

```diff
@@ -92,7 +98,16 @@ func checkForPipelineResourceDuplicates(resources []TaskResourceBinding, path st
 			return apis.ErrMultipleOneOf(path)
 		}
 		encountered[name] = struct{}{}
+		// Check that one of resource ref and resource Spec is present
+		if r.ResourceRef.Name != "" && r.ResourceSpec != nil {
+			return apis.ErrDisallowedFields(fmt.Sprintf("%s.ResourceRef", path), fmt.Sprintf("%s.ResourceSpec", path))
+		}
+		// Check that one of resource ref and resource Spec is present
+		if r.ResourceRef.Name == "" && r.ResourceSpec == nil {
+			return apis.ErrMissingField(fmt.Sprintf("%s.ResourceRef", path), fmt.Sprintf("%s.ResourceSpec", path))
+		}
```

Comment:
should we call something here to validate the spec itself?

Reply:
Yeah thats a good idea. I will add that